src/model/codacyInput.ts

```diff
@@ -12,7 +12,7 @@ export class Parameter {
 
 export class Pattern {
   readonly patternId: string
-  readonly parameters: Parameter[]
+  parameters: Parameter[]
```

Comment:
Consider leaving the class immutable and create a copy when you need to add the other parameters.